### PR TITLE
Reflect channel creating permissions on UI. + simple channel creation error notifications

### PR DESCRIPTION
**Pull request by Fajfa**
Prevented creating public/group/private channels based on effectivePermissions. 
If private channel creation is disabled, user cant click to create new private conversations. 

Forced channel creation form reload on channel type switch. 

Error when creating channel: 
Screenshot from 2019-07-05 12-53-38

### CODE REVIEW

File: src/views/ChannelEditor.vue [hunk 1]
```diff
@@ -190,7 +185,7 @@ export default {
     },
 
     'type' (newType) {
-      if (!this.channel.channelID) this.channel.type = newType
+      if (!this.channel.channelID) this.load()
```
Comment (darh) [inline src/views/ChannelEditor.vue:202]: Why this change?

File: src/components/Panel/Channels/index.vue [hunk 2]
```diff
@@ -140,6 +144,16 @@ export default {
     },
   },
 
+  created () {
+    this.$MessagingAPI.permissionsEffective().then(e => {
```
Comment (darh) [inline src/components/Panel/Channels/index.vue]: Move API comm out of components into views and use vuex.
In messaging, we will move a couple of other things there as well so let's start preparing a good environment for that.

Lets's build a vuex store (`session` looks like a good name for it) that will keep the stuff we need for current session. Permissions are one of the the first thing and adding an object of actions (current) user is allowed to do (`{ channelGroupCreate: bool }`). Add it under `permissions`.
Add loading of it under `init` action and add all mutations we need. Base upon other, existing stores (`pending` state).

Then add (and use) simple getters (`canCreateGroupChannel()`...).

**Review by darh — CHANGES_REQUESTED** [review]
1. You are solving 2 issues so it would be better to split this into 2 PRs.
2. Implementation for permissions should be more effective. I've left a comment about it and how we should move it to vuex. 
3. Are final errors more descriptive that one shown on screenshot?

File: src/views/ChannelEditor.vue (hunk 1 above)
Comment (Fajfa) [reply src/views/ChannelEditor.vue:202]: Noticed a bug where the form doesnt switch on type change, public/private/group.
Steps to reproduce on latest:
Click the '+' next to direct messages. 
Then click the '+' next to public channels.
The form wont switch from group channel creation to public channel creation.

This fixes that.

File: src/components/Panel/Channels/index.vue (hunk 2 above)
Comment (Fajfa) [reply src/components/Panel/Channels/index.vue]: Deal. This is the next step.

**Comment by Fajfa** [comment]
The error message is just a sample. For now the error from api goes through our translation system, so we can add the translations for the possible errors. Still TODO(not sure of the syntax of the errors possible)

**Comment by darh** [comment]
Where are we with this @Fajfa ?

**Comment by Fajfa** [comment]
After rebase, ready to merge. Been done for some time now, i kinda forgot about it to be honest.

**Comment by darh** [comment]
Merge then.
